magento/magento2
# Fix Adapter\Pdo\Mysql::closeConnection on non-default ports

### Description (*)

Fixing Magento\Framework\DB\Adapter\Pdo\Mysql::closeConnection when the port is a non-default port.  When we are using a non-default port for the database,  and closeConnection() method is called,  this prevents automatic reconnection from working.  This is because during the connection,  we move the port into $this->_config['port'].

The closeConnection() is not called in normal Magento usage.  However,  it is always called when running the integration tests.  This means that that you cannot use integration tests with a Mysql server on a non-default port.


### Fixed Issues (if relevant)



### Manual testing scenarios (*)


Steps to reproduce the error that this bugs fixes:

1. Configure a Mysql server using a custom port.
2. Configure Magento integration tests to use this database.
3. Try to run integration tests.

Expected Result:  Integration tests pass.

Actual Result:  Integration tests fail with errors because they use closeConnection() which loses the port configuration.


### Questions or comments


### Contribution checklist (*)
